### PR TITLE
[WFCORE-6371][WFCORE-6393][WFCORE-6394] Adding support for o.j.m.s.ServiceTarget.addService() method plus deprecated methods cleanup

### PR DESCRIPTION
**Pull request by ropalka**
https://issues.redhat.com/browse/WFCORE-6371
https://issues.redhat.com/browse/WFCORE-6393
https://issues.redhat.com/browse/WFCORE-6394

### CODE REVIEW

**Review by yersan — APPROVED** [review]

File: controller/src/main/java/org/jboss/as/controller/AbstractOperationContext.java [hunk 1]
```diff
@@ -1402,7 +1401,7 @@ class Step extends AbstractStep {
         private ResultHandler resultHandler;
         ServiceTarget serviceTarget;
         private ServiceVerificationHelper serviceVerificationHelper;
-        private Set<ServiceName> addedServices;
+        private Set<ServiceController<?>> addedServices;
```
Comment (pferraro) [inline controller/src/main/java/org/jboss/as/controller/AbstractOperationContext.java:1404]: The backing set (created in getAddedServices()) is a HashSet, but ServiceControllerImpl does not implement hashCode semantics, nor does DelegatingServiceController delegate hashCode()/equals() to the underlying ServiceController.  That seems problematic.

**Review by pferraro — CHANGES_REQUESTED** [review]

File: controller/src/main/java/org/jboss/as/controller/AbstractOperationContext.java (hunk 1 above)
Comment (ropalka) [reply controller/src/main/java/org/jboss/as/controller/AbstractOperationContext.java:1405]: Good catch, thanks @pferraro ! It is fixed now.

File: controller/src/main/java/org/jboss/as/controller/OperationContextImpl.java [hunk 2]
```diff
@@ -146,7 +146,7 @@ final class OperationContextImpl extends AbstractOperationContext implements Aut
 
     private final ModelControllerImpl modelController;
     private final OperationMessageHandler messageHandler;
-    private final Map<ServiceName, ServiceController<?>> realRemovingControllers = new HashMap<>();
+    private final Set<ServiceController<?>> realRemovingControllers = new HashSet<>();
```
Comment (pferraro) [inline controller/src/main/java/org/jboss/as/controller/OperationContextImpl.java:149]: This should also use Collections.newSetFromMap(new IdentityHashMap<>()), no?

**Review by pferraro — CHANGES_REQUESTED** [review]

File: controller/src/main/java/org/jboss/as/controller/OperationContextImpl.java (hunk 2 above)
Comment (yersan) [reply controller/src/main/java/org/jboss/as/controller/OperationContextImpl.java:149]: @ropalka What's your opinion about Paul's comment?

Comment (ropalka) [reply controller/src/main/java/org/jboss/as/controller/OperationContextImpl.java]: Correct @pferraro !

**Review by pferraro — APPROVED** [review]

**Comment by yersan** [comment]
Thanks @ropalka, @pferraro

**Pull request merged**